src/main/java/jenkins/scm/api/mixin/ChangeRequestSCMHead.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package jenkins.scm.api.mixin;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import jenkins.scm.api.SCMHead;
+import jenkins.scm.api.SCMSource;
+import org.kohsuke.stapler.export.Exported;
+
+/**
+ * Mixin interface to identify {@link SCMHead} instances that correspond to a change request.
+ *
+ * @since 2.0
+ */
+public interface ChangeRequestSCMHead extends SCMHeadMixin {
+
+    /**
+     * Identifier of this change request.
+     * Expected to be unique among requests coming from a given {@link SCMSource}.
+     *
+     * @return an ID of some kind, such as a pull request number (decimal) or a Gerrit change ID
+     */
+    @Exported
+    @NonNull
+    String getId();
+
+    /**
+     * Branch to which this change would be merged or applied if it were accepted.
```

Comment:
ACK. *if* SCMHead had been retained as a pure String wrapper that just adds usage type information (and perhaps tracking of origin based on the SCMSource) then there would be no reason. 

Once GitHub and BitBucket started storing the Target in the SCMHead (and not in the SCMRevision) then we need this.

Keep in mind that the SCMRevision equality is based on the revisions's head, so if the Target changes, with this equals we get the revision always being changed and getting a rebuild... without it, for revisions that are just tracking the fork (but untrusted and using the Jenkinsfile from the target) will not get their rebuild... we could fix that by moving the target to SCMRevision, but the ship has sailed due to the usage in those two plugins and I see some additional utility in making the target be part of the head identity